### PR TITLE
Fix list pick not working in remotes

### PR DESCRIPTION
**Pull request by ItsTisme**
## About The Pull Request

during my original implementation. I made a mistake in adjescent which means it did not work in remotes and BMI's, this should fix it

## Why It's Good For The Game

A bug on my part, it can be used in remotes now(as it should have from the start)

## Changelog
:cl:
fix: list pick works in remotes/BMI's now
/:cl:

### CODE REVIEW

File: code/modules/wiremod/components/list/list_pick.dm [hunk 1]
```diff
@@ -54,24 +54,24 @@
...
+	if(isliving(user.value))
+		var/mob/living/living_user = user.value
+		if(!(living_user.can_perform_action(parent.shell, FORBID_TELEKINESIS_REACH, ALLOW_SILICON_REACH, ALLOW_RESTING)))
+			return
 		trigger_output.set_output(COMPONENT_SIGNAL)
 		INVOKE_ASYNC(src, PROC_REF(show_list), user.value, input_name.value, input_list.value)
 
 /// Show a list of options to the user using standed TGUI input list
-/obj/item/circuit_component/list_pick/proc/show_list(mob/user, message, list/showed_list)
+/obj/item/circuit_component/list_pick/proc/show_list(mob/living/u, message, list/showed_list)
```
Comment (Watermelon914) [inline code/modules/wiremod/components/list/list_pick.dm:65]: I'm not sure why you're changing user to a single letter variable name but it removes information, so please revert it.

**Review by Watermelon914 — CHANGES_REQUESTED** [review]
If you are going to make any modifications to the list pick component, I think it's important that you rework it into a shell.

There's too many problems with forcing a prompt onto players via circuits, it's not really what I want out of a circuit component specifically.

Lmk if you don't want to do this, since I'll probably just end up doing it myself.

You're free to make a BCI exclusive for this though, as long as user is hardlocked to whoever the BCI is inside of.

File: code/modules/wiremod/components/list/list_pick.dm (hunk 1 above)
Comment (Autisem) [reply code/modules/wiremod/components/list/list_pick.dm:65]: Was more to imply/ensure it was from LIVING and not from MOB. also I have a circuit input named  user so name conflict happend which meant it was not clear which user he used

**Comment by Autisem** [comment]
>If you are going to make any modifications to the list pick component, I think it's important that you rework it into a shell.

I would like though to have one as a standing machine, BMI or a remote. I could make 3 shells but i felt a component would just add more freedom to add it to any circuit(say you wana make a circuit based vending machine, needs dispenser or money shell. which a new shell would make needlessly complicated/impossible)

File: code/modules/wiremod/components/list/list_pick.dm (hunk 1 above)
Comment (Autisem) [reply code/modules/wiremod/components/list/list_pick.dm]: I made the name a bit clearer, but making it user would be not great due to the mentioned conflict issue

**Comment by Watermelon914** [comment]
The problem is primarily due to the fact that you can force a prompt onto players via a circuit component directly.
Although it's more annoying, the circuit should not be able to decide who is currently picking the list, that needs to be provided to them instead. Otherwise, we have the problem of players easily being able to make disruptive circuits that force a prompt onto players, which shouldn't really be possible.

If you have any better ideas on how to solve this problem, feel free to propose them, but the current design goes against what independent circuit components are supposed to be able to do.

File: code/modules/wiremod/components/list/list_pick.dm (hunk 1 above)
Comment (Watermelon914) [reply code/modules/wiremod/components/list/list_pick.dm]: could name it something more detailed like `showing_to`

**Comment by Watermelon914** [comment]
> I could make 3-4 different shells to do EXACT same thing in different form factors but at that point why even have components.

Shells are there to primarily handle the inputs, components can usually take inputs too but not directly from users.
I wouldn't be opposed to some sort of implementation that lets you extend specific shell capabilities.

I.e. bot shells, handhelds and the such being able to be tweaked to take a different input instead of just clicking.

**Comment by Autisem** [comment]
> I wouldn't be opposed to some sort of implementation that lets you extend specific shell capabilities.

Maby some extention module, where only 1 or few fit. bit I have to brain juice on how to actuely make that without it being tedious and/or shit code

**Comment by Autisem** [comment]
watermelon will be doing this so ill close this

**Pull request closed without merging**